examples/clear_and_rebuild_index.rs

```diff
@@ -0,0 +1,63 @@
+// # Clearing and rebuilding an index
+//
+// This example shows how to build, clear and rebuild an index
+
+// Importing tantivy...
+#[macro_use]
+extern crate tantivy;
+use tantivy::collector::TopDocs;
+use tantivy::query::QueryParser;
+use tantivy::schema::*;
+use tantivy::Index;
+
+fn main() -> tantivy::Result<()> {
+    // We first create a schema for the sake of the
+    // example. Check the `basic_search` example for more information.
+    let mut schema_builder = Schema::builder();
+
+    // For this example, we need to make sure to index positions for our title
+    // field. `TEXT` precisely does this.
+    let title = schema_builder.add_text_field("title", TEXT | STORED);
+    let schema = schema_builder.build();
+
+    let index = Index::create_in_ram(schema.clone());
+
+    let mut index_writer = index.writer_with_num_threads(1, 50_000_000)?;
+    index_writer.add_document(doc!(title => "The Old Man and the Sea"));
+    index_writer.add_document(doc!(title => "The modern Promotheus"));
+    index_writer.commit()?;
+
+    // clears the index
+    index_writer.clear()?;
+    // have to commit, otherwise deleted terms remain available
+    index_writer.commit()?;
```

Comment:
@fulmicoton @torkleyy 

Should clear have an implicit/hidden commit at the end of it or should we make users commit manually? 

I am leaning towards hiding/doing the commit for the caller to help them fall into the [pit of success](https://blog.codinghorror.com/falling-into-the-pit-of-success/) instead of forgetting to do it and finding old files.

Reply:
I don't think we need an example for that one. We could have a code sample in the rustdoc though.

Reply:
The benefit of having an extra commit step is that people is :
- you can write something where no user will ever see the state where your index was empty.
- you can write code that removes all segments, adds new documents (typically a full reindexing) coming from an external source. (e.g. a KV store like Cassandra for instance). If this external source fails, you can abort the process without having lost everything.

I think requiring a commit would be nice.